cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/core/mongo/impl/SearchProviderRegistry.java

```diff
@@ -0,0 +1,41 @@
+package edu.asu.diging.cord19.explorer.core.mongo.impl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.cord19.explorer.web.model.SearchType;
+
+@Service
+public class SearchProviderRegistry {
+
+    @Autowired
+    private List<SearchProvider> searchProviders;
+
+    private Map<SearchType, String> views = new HashMap<>();
+
+    private Map<SearchType, SearchProvider> map = new HashMap<>();
+
+    @PostConstruct
+    public void init() {
+        for (SearchProvider search : searchProviders) {
+            map.put(search.getSearchType(), search);
+        }
+        views.put(SearchType.AFFILIATIONS, "affResults");
+        views.put(SearchType.PUBLICATIONS, "results");
```

Comment:
this is view logic and should not be in your business logic but in your controller